lexos/helpers/general_functions.py

```diff
@@ -285,32 +197,38 @@ def apply_function_exclude_tags(text, functions):
     for function_to_apply in functions:
         contents[-1] = function_to_apply(contents[-1])
     striped_text += contents[-1]
-
     return striped_text
 
 
-def decode_bytes(raw_bytes):
-    """
-    decode the raw bytes, typically used to decode `request.file`
-    Args:
-        raw_bytes: the bytes you get and want to decode to string
-    """
-    try:
-        # try to use utf-8 to decode first
-        encoding_type = "utf-8"
-        # Grab the file contents, which were encoded/decoded automatically into
-        # python's format
-        decoded_file_string = raw_bytes.decode(encoding_type)
-
-    except UnicodeDecodeError:
-
-        encoding_detect = chardet.detect(
-            raw_bytes[:constants.MIN_ENCODING_DETECT])  # Detect the encoding
+def decode_bytes(raw_bytes: bytes) -> str:
+    """decode the raw bytes, typically used to decode `request.file`
 
-        encoding_type = encoding_detect['encoding']
-
-        # Grab the file contents, which were encoded/decoded automatically into
-        # python's format
-        decoded_file_string = raw_bytes.decode(encoding_type)
-
-    return decoded_file_string
+    :param raw_bytes: the bytes you get and want to decode to string
+    :return: A decoded string
+    """
+    bom16_le = str(codecs.BOM_UTF16_LE)
+    bom16_be = str(codecs.BOM_UTF16_BE)
+    if str(raw_bytes).startswith(bom16_le):
+        decoded_string = raw_bytes[len(bom16_le):-1]
+        decoded_string = decoded_string.replace("b'", "")
+        decoded_string = decoded_string.replace("\\x00", "")
+    elif str(raw_bytes).startswith(bom16_be):
+        decoded_string = raw_bytes[len(bom16_be):-1]
+        decoded_string = decoded_string.replace("b'", "")
+        decoded_string = decoded_string.replace("\\x00", "")
```

Comment:
this is extremely unsafe. directly removing bytes from the string can easily destroy the information in string.

Please investigate more on why `chardet` are decoding `utf-16` bytes as `utf-8`